sfputil/debug.py

```diff
@@ -0,0 +1,123 @@
+import sys
+import click
+import utilities_common.cli as clicommon
+from utilities_common import platform_sfputil_helper
+from utilities_common.platform_sfputil_helper import (
+    get_subport,
+    get_sfp_object,
+    get_value_from_db_by_field,
+    get_subport_lane_mask
+)
+
+EXIT_FAIL = -1
+EXIT_SUCCESS = 0
+ERROR_PERMISSIONS = 1
+ERROR_CHASSIS_LOAD = 2
+ERROR_SFPUTILHELPER_LOAD = 3
+ERROR_PORT_CONFIG_LOAD = 4
+ERROR_NOT_IMPLEMENTED = 5
+ERROR_INVALID_PORT = 6
+
+
+@click.group(cls=clicommon.AliasedGroup)
+def debug():
+    """
+    Group for debugging and diagnostic control commands.
+
+    This command group loads platform-specific utilities and prepares them for use in diagnostic commands.
+    """
+    platform_sfputil_helper.load_platform_sfputil()
+    platform_sfputil_helper.load_chassis()
+    platform_sfputil_helper.platform_sfputil_read_porttab_mappings()
+
+
+@debug.command()
+@click.argument('port_name', required=True)
+@click.argument(
+    'loopback_mode',
+    required=True,
+    type=click.Choice(["host-side-input", "host-side-output", "media-side-input", "media-side-output"])
+)
+@click.argument('enable', required=True, type=click.Choice(["enable", "disable"]))
+def loopback(port_name, loopback_mode, enable):
+    """
+    Set module diagnostic loopback mode.
+    """
+    sfp = get_sfp_object(port_name)
+
+    try:
+        api = sfp.get_xcvr_api()
+    except NotImplementedError:
+        click.echo(f"{port_name}: This functionality is not implemented")
+        sys.exit(ERROR_NOT_IMPLEMENTED)
+
+    subport = get_subport(port_name)
+
+    host_lane_count = get_value_from_db_by_field("STATE_DB", "TRANSCEIVER_INFO", "host_lane_count", port_name)
+
+    media_lane_count = get_value_from_db_by_field("STATE_DB", "TRANSCEIVER_INFO", "media_lane_count", port_name)
+
```

Comment:
@vdahiya12 one function each for get_media_lane_count() and get_host_lane_count() for future use as well?

Reply:
fixed